src/pages/history/history.js

```diff
@@ -1,18 +1,31 @@
 import styles from "./history.module.scss"
 import { VideoCard, SidePane } from "@components"
 import { Button } from "react-felix-ui"
+import { useHistory } from "@providers"
+
 const History = () => {
+    const { HistoryState: { history }, removeFromHistory, clearHistory } = useHistory()
+
     return (
         <div className={styles.container}>
             <SidePane title="History" time="30 Minutes" count={4} >
-                <Button variant="ghost" isRound={true}>Clear History</Button>
+                <Button variant="ghost" isRound={true} onClick={clearHistory}>Clear History</Button>
             </SidePane>
             <div className={styles.main}>
-                {/* <VideoCard orientation="horizontal" />
-                <VideoCard orientation="horizontal" />
-                <VideoCard orientation="horizontal" />
-                <VideoCard orientation="horizontal" />
-                <VideoCard orientation="horizontal" /> */}
+                {
+                    history.length !== 0
+                        ? [...history].reverse().map((item, i) => {
+                            return <VideoCard
+                                orientation="horizontal"
+                                videoItem={item}
+                                key={i}
+                                onRemove={removeFromHistory}
+                            />
+                        })
+                        : [...Array(4)].map((_, i) => {
```

Comment:
when this will timeout?, right now it looks loading forever

Reply:
The state changes for map once the data is updated, there is a bug for infinite loading. will be fixing that soon. thanks for noticing.